diff --git a/ios/Worklets/EXJavaScriptSerializable.h b/ios/Worklets/EXJavaScriptSerializable.h
index e5547d94c1bae560940c0350eeb5573d5913b159..0a629372c7cc04393a97395a9443719f4984f8d9 100644
--- a/ios/Worklets/EXJavaScriptSerializable.h
+++ b/ios/Worklets/EXJavaScriptSerializable.h
@@ -26,6 +26,8 @@ typedef NS_ENUM(NSInteger, EXSerializableValueType) {
   EXSerializableValueTypeImport = 17,
   EXSerializableValueTypeSynchronizable = 18,
   EXSerializableValueTypeCustom = 19,
+  EXSerializableValueTypeSymbol = 20,
+  EXSerializableValueTypeShareable = 21,
 } NS_SWIFT_NAME(SerializableValueType);
 
 NS_SWIFT_NAME(JavaScriptSerializable)
diff --git a/ios/Worklets/EXJavaScriptSerializable.mm b/ios/Worklets/EXJavaScriptSerializable.mm
index 93bf657a549db0053f3d25026de996e8619f3779..d1dc05a66f0abbfc9d952682c771b87a3f2afec8 100644
--- a/ios/Worklets/EXJavaScriptSerializable.mm
+++ b/ios/Worklets/EXJavaScriptSerializable.mm
@@ -48,6 +48,10 @@ EXSerializableValueType toObjCValueType(worklets::Serializable::ValueType type)
       return EXSerializableValueTypeSynchronizable;
     case worklets::Serializable::ValueType::CustomType:
       return EXSerializableValueTypeCustom;
+    case worklets::Serializable::ValueType::SymbolType:
+      return EXSerializableValueTypeSymbol;
+    case worklets::Serializable::ValueType::ShareableType:
+      return EXSerializableValueTypeShareable;
   }
 }
 
